### ISSUE TITLE
decoupleR::get_collectri 403 error

### ISSUE THREAD

**Issue by mfranke-2** [saezlab/CollecTRI#19]
Hi, I have been using CollecTRI successfully until recently, when I started receiving the following error:

`decoupleR::get_collectri(organism="human" split_complexes=FALSE)`
> [2024-06-04 21:20:36] [WARN]    [OmnipathR] HTTP 403
[2024-06-04 21:20:36] [WARN]    [OmnipathR] Failed to download "https://www.ensembl.org/info/about/species.html" (attempt 1/3); error: HTTP 403
[2024-06-04 21:20:42] [WARN]    [OmnipathR] HTTP 403
[2024-06-04 21:20:42] [WARN]    [OmnipathR] Failed to download "https://www.ensembl.org/info/about/species.html" (attempt 2/3); error: HTTP 403
[2024-06-04 21:20:47] [WARN]    [OmnipathR] HTTP 403
[2024-06-04 21:20:47] [ERROR]   [OmnipathR] Failed to download "https://www.ensembl.org/info/about/species.html" (attempt 3/3); error: HTTP 403
Error in 'map_int()':
In index: 1.
Caused by error in 'map_int()':
In index: 1.
Caused by error:
! HTTP 403

Any help would be greatly appreciated!

Best,
Megan

> sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:
[1] dplyr_1.1.4      dorothea_1.14.1  OmnipathR_3.13.4 decoupleR_2.9.7

**Comment by smuellerd** [comment]
@deeenes, could you have a look at this?

**Comment by mfranke-2** [comment]
Hi! I just wanted to follow up on this - It seems that the ensembl server is checking for certain headers to determine if it's a browser request or request from code (and it's blocking requests from code). Maybe adding a header argument when reading the .html file could fix this?

**Comment by deeenes** [comment]
@mfranke-2 Thanks for the tip, I'll look into it! Though now I managed to download it with the default headers of cli `curl`. There is something weird going on with ensembl.org, since the update of their ssl certificate last month the CI under OSX fails. It's quite possible the two issues are the same.

Did you experience the issue in different times, on different networks and computers, or it happened only once? Have you tried it again since then?

**Comment by mfranke-2** [comment]
Interesting, yes I think you're right that the two issues are related or the same! In R, something as simple as the following yields the 403 error for me:
`download.file("https://useast.ensembl.org/info/about/species.html", destfile = "test.txt", headers = c("User-Agent" = "My Custom User Agent"))`

but changing the header argument fixes the issue:

`download.file("https://useast.ensembl.org/info/about/species.html", destfile = "test.txt", headers = c("User-Agent" = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.5 Safari/605.1.15"))`

I've tried it on two different computers and received the same error. I also tried to run again this morning and again had the error.
